libraries/CurieBLE/examples/IMUBleCentral/IMUBleCentral.ino

```diff
@@ -0,0 +1,152 @@
+typedef struct {
+    int index;
+    unsigned int slot[3];
+} imuFrameType;
+
+imuFrameType imuBuf[MAX_IMU_RECORD];
+
+unsigned seqNum = 0;
+
+long previousMillis = 0;  // last time the battery level was checked, in ms
+
+BLECentral bleCentral;;       // BLE Central Device (the board you're programming)
+
+BLEService bleImuService("F7580001-153E-D4F6-F26D-43D8D98EEB13");
+// Tx IMU data Characteristic
+BLECharacteristic bleImuChar("F7580003-153E-D4F6-F26D-43D8D98EEB13",       // standard 16-bit characteristic UUID
+                             BLERead | BLENotify, sizeof(imuBuf));     // remote clients will be able to
+                                               // get notifications if this characteristic changes
+
+void ble_connected(BLEHelper &role)
+{
+    BLEPeripheralHelper&peripheral = *(BLEPeripheralHelper*)(&role);
+    Serial.println("Connected");
+    
+    // Start discovery the profiles in peripheral device
+    peripheral.discover();
+}
+
+void bleImuCharacteristicWritten(BLEHelper& central, BLECharacteristic& characteristic)
+{
+    // Peripheral wrote new value to characteristic by Notification/Indication
+    const unsigned char *cvalue = characteristic.value();
+    const imuFrameType *value = (const imuFrameType *)cvalue;
+    Serial.print("\r\nCharacteristic event, written: ");
+    Serial.print(value->index);
+    Serial.print("\t");
+    Serial.print(value->slot[0]);
+    Serial.print("\t");
+    Serial.print(value->slot[1]);
+    Serial.print("\t");
+    Serial.println(value->slot[2]);
+    Serial1.print("\r\n");
+}
+
+bool adv_found(uint8_t type, 
+               const uint8_t *data, 
+               uint8_t data_len,
+               void *user_data)
+{
+    bt_addr_le_t *addr = (bt_addr_le_t *)user_data;
+    int i;
+
+    Serial.print("[AD]:");
+    Serial.print(type);
+    Serial.print(" data_len ");
+    Serial.println(data_len);
+
+    switch (type)
+    {
+        case BT_DATA_UUID128_SOME:
+        case BT_DATA_UUID128_ALL:
+        {
+            if (data_len % MAX_UUID_SIZE != 0) 
+            {
+                Serial.println("AD malformed");
+                return true;
+            }
+            struct bt_uuid * serviceuuid = bleImuService.uuid();
+            for (i = 0; i < data_len; i += MAX_UUID_SIZE)
+            {
+                if (memcmp (((struct bt_uuid_128*)serviceuuid)->val, &data[i], MAX_UUID_SIZE) != 0) 
+                {
+                    continue;
+                }
+                
+                // Accept the advertisement
+                if (!bleCentral.stopScan()) 
+                {
+                    Serial.println("Stop LE scan failed");
+                    continue;
+                }
+                Serial.println("Connecting");
+                // Connect to peripheral
+                bleCentral.connect(addr, &conn_param);
+                return false;
+            }
+        }
+    }
+
+    return true;
+}
+
+void setup() {
+    Serial1.begin(115200);
```

Comment:
Why are both Serial and Serial1 needed for this example sketch? can't you just print all the information to one serial port?